features/storage/kvstore/tdbstore/TDBStore.cpp

```diff
@@ -413,6 +413,8 @@ int TDBStore::set_start(set_handle_t *handle, const char *key, size_t final_data
         // in the upper layers).
         ih->bd_base_offset = _master_record_offset;
         ih->new_key = false;
+        ram_table_ind = 0;
+        hash = 0;
```

Comment:
It would be simpler to initialise these variables with default values where they are declared.

Reply:
No.
This would hide if there are code paths that do not initialise these values. This one is one of the example.

If we initialise all local variables in beginning of function, then Coverity is not able to tell that we use then un-initialised in certain use-case.

Reply:
okay